libs/application/template-api-modules/src/lib/modules/templates/p-sign-submission/p-sign-submission.service.ts

```diff
@@ -61,4 +95,48 @@ export class PSignSubmissionService {
     }
     return { success: result.success, id: result.caseNumber }
   }
+
+  private getName(application: Application): string {
+    const nationalRegistryData = application.externalData.nationalRegistry
+      ?.data as NationalRegistry
+    const dateStr = new Date(Date.now()).toISOString().substring(0, 10)
+
+    return `p_kort_mynd_${nationalRegistryData?.nationalId}_${dateStr}.pdf`
+  }
+
+  private async getAttachments({
+    application,
+    auth,
+  }: TemplateApiModuleActionProps): Promise<string> {
+    const attachments = getValueViaPath(
+      application.answers,
+      'attachments',
+    ) as Array<{ key: string; name: string }>
+    const hasAttachments = attachments && attachments?.length > 0
+
+    if (!hasAttachments) {
+      return Promise.reject({})
+    }
+
+    const { key } = attachments[0]
+
+    const contentData = await this.sharedTemplateAPIService
+      .makeGraphqlQuery<ContentData>(auth.authorization, QUERY, {
+        input: {
+          id: application.id,
+          key: key,
+        },
+      })
+      .then((response) => response.json())
```

Comment:
This code should use the fileService to fetch the file contents directly from S3 instead of taking a loop through GraphQL:

ApplicationSystemAPI -> GraphQL API -> ApplicationSystemAPI -> S3.